### PR TITLE
switch to Amazon ECR public registry for image pulls

### LINKED ISSUE #1380: Switch to Amazon ECR public registry
We started hitting rate limits with Docker Hub: https://github.com/LibertyDSNP/frequency/actions/runs/4672262948/jobs/8274250371

Consider pulling Ubuntu images from Amazon ECR public registry instead where rate limiting is much more relaxed. For example:

```
$ docker pull public.ecr.aws/lts/ubuntu:20.04
```

Here is a good summary of alternative options: https://stackoverflow.com/a/73512606/2619188

### PR DESCRIPTION
**Pull request by demisx**
# Goal
The goal of this PR is to switch to Amazon ECR public registry for Ubuntu image pulls instead of heavily rate limited Docker Hub.

Closes #1380

### CODE REVIEW

**Review by johnmela — APPROVED** [review]

File: .github/workflows/verify-pr-commit.yml [hunk 1]
```diff
@@ -463,15 +460,14 @@ jobs:
...
-    container: ubuntu:20.04
+    container: public.ecr.aws/lts/ubuntu:20.04
     steps:
       - name: Install Required Packages
         run: |
           apt-get update
           apt-get install -y curl
           curl -fsSL https://get.docker.com -o get-docker.sh
           sh get-docker.sh
-          docker run hello-world
```
Comment (johnmela) [inline .github/workflows/verify-pr-commit.yml:474]: Was this removal intended?

**Review by saraswatpuneet — APPROVED** [review]

File: .github/workflows/verify-pr-commit.yml (hunk 1 above)
Comment (wilwade) [reply .github/workflows/verify-pr-commit.yml:474]: It looks like just removing a check that docker is working.  @demisx ?
